sdk/javascript/generator/TypeFormatter.py

```diff
@@ -65,10 +65,17 @@ def generate_ctor(self):
 		method_descriptor.method_name = 'constructor'
 		return self.generate_method(method_descriptor)
 
-	def generate_deserializer(self):
-		# 'deserialize'
-		method_descriptor = self.provider.get_deserialize_descriptor()
-		method_descriptor.method_name = 'static deserialize'
+	def generate_deserializer(self, name, generated_name=None):
+		if not hasattr(self.provider, f'get_{name}_descriptor'):
+			return None
+
+		descriptor_getter = getattr(self.provider, f'get_{name}_descriptor')
+		method_descriptor = descriptor_getter()
+		if not method_descriptor:
+			return None
```

Comment:
when can this return None when function exists (previous check passes)?

Reply:
if you have `get_deserialize_aligned_descriptor` that returns None-like thing
i.e. for pod types:
```
	def get_deserialize_aligned_descriptor(self):
		if self._is_array:
			return None
```